web-wallet/src/lib/containers/MigrateContract/MigrateContract.svelte

```diff
@@ -1,133 +1,196 @@
+  import {
+    AppAnchor,
+    AppAnchorButton,
+    AppImage,
+    ApproveMigration,
+    ExecuteMigration,
+  } from "$lib/components";
   import {
     Button,
     ExclusiveChoice,
     Icon,
+    Stepper,
     Textbox,
-    Wizard,
-    WizardStep,
   } from "$lib/dusk/components";
-  import {
-    mdiArrowRight,
-    mdiCheckDecagramOutline,
-    mdiInformationOutline,
-    mdiTimerSand,
-    mdiWalletOutline,
-  } from "@mdi/js";
-  import { createCurrencyFormatter } from "$lib/dusk/currency";
   import { logo } from "$lib/dusk/icons";
-  import { calculateAdaptiveCharCount, middleEllipsis } from "$lib/dusk/string";
-  import { onDestroy, onMount } from "svelte";
-  import { settingsStore } from "$lib/stores";
+  import { settingsStore, walletStore } from "$lib/stores";
   import {
     account,
-    accountBalance,
     modal,
     wagmiConfig,
     walletDisconnect,
   } from "$lib/migration/walletConnection";
-  import { switchChain } from "@wagmi/core";
-  import { bsc, mainnet } from "viem/chains";
+  import { allowance, balanceOfCoin } from "$lib/migration/migration";
 
+  $: ({ address, chainId, isConnected } = $account);
+  $: ({ currentAddress } = $walletStore);
+  $: isAmountValid =
+    amount >= minAmount && amount <= Number(connectedWalletBalance);
+
+  const { darkMode, network: migrationNetwork } = $settingsStore;
+
+  /** @type {"mainnet" | "testnet"} */
+  // @ts-ignore
+  const network = migrationNetwork.toLowerCase();
+
+  /** @type {Tokens} */
   const tokens = {
-    bnb: {
-      chainId: bsc.id,
-      name: "BEP-20",
+    mainnet: {
+      "BEP-20": {
+        chainId: bsc.id,
+        contract: "0xb2bd0749dbe21f623d9baba856d3b0f0e1bfec9c",
+        migrationContract: "0x",
+        name: "BEP-20",
+      },
+      "ERC-20": {
+        chainId: mainnet.id,
+        contract: "0x940a2db1b7008b6c776d4faaca729d6d4a4aa551",
+        migrationContract: "0x",
+        name: "ERC-20",
+      },
     },
-    eth: {
-      chainId: mainnet.id,
-      name: "ERC-20",
+    testnet: {
+      "BEP-20": {
+        chainId: sepolia.id,
+        contract: "0xC416f5d2AE6BAec2a23f412Df11166afC35CAba2",
+        migrationContract: "0x9f5d1c067710fc6ed49a6444afd69b64799a57b6",
+        name: "BEP-20",
+      },
+      "ERC-20": {
+        chainId: sepolia.id,
+        contract: "0x92DA9BE2039E818bB78223A6BA7C85CC2b17D8D5",
+        migrationContract: "0x63fd2B12034e108BCe73e8832b7dabC8bd67f738",
+        name: "ERC-20",
+      },
     },
   };
 
   const options = [
-    { disabled: false, label: tokens.eth.name, value: tokens.eth.name },
-    { disabled: false, label: tokens.bnb.name, value: tokens.bnb.name },
+    {
+      disabled: false,
+      label: tokens[network]["ERC-20"].name,
+      value: tokens[network]["ERC-20"].name,
+    },
+    {
+      disabled: false,
+      label: tokens[network]["BEP-20"].name,
+      value: tokens[network]["BEP-20"].name,
+    },
   ];
```

Comment:
Could refactor using a map:

```
const tokenTypes = ["ERC-20", "BEP-20"];

const options = tokenTypes.map(type => ({
  disabled: false,
  label: tokens[network][type].name,
  value: tokens[network][type].name,
}));
```

Reply:
if we always use the `name` for both label and value you can just pass the array of strings